Fix Linux upload path detection

BigWhale opened the pull request:
### Requirements

None

### Description

Detecting a correct upload drive for LPC1768 on Ubuntu Linux is broken. When an external drive is connected it will be mounted in /media/<username>/<drive>. Right now upload_extra_script.py only checks the /media directory for possible drives. It will always fail since all the drives are mounted in the users media directory.

Also, there's a lot of string concatenation going and I replaced it with os.path.join which is safer to use.


### Benefits

Drive detection will work and the built binary will be copied to the drive.

### Related Issues
None

Commit:
Corrected upload path detection for linux

thinkyhead commented:
Thanks for your contribution! Unfortunately we can't accept PRs directed at release branches. We make patches to the bugfix branches and only later do we push them out as releases. Please redo this PR starting with the `bugfix-2.0.x` branch and be careful to target `bugfix-2.0.x` when resubmitting the PR.

It may help to set your fork's default branch to `bugfix-2.0.x`.

See http://marlinfw.org/docs/development/getting_started_pull_requests.html for full instructions.

ellensp commented:
Never had this issue....  what version of ubuntu?  Description also doesn't make much sense...  
old code checks all sub directories in /media/ ie goes through each users directories looking for REARM   new code assembles the directory "/"+"media"+"/"+{username}  and checks for REARM...  This is the same thing with less directory trawling...

thinkyhead commented:
Rebased. Sorry, we had to screw up your `2.0.x` branch to do it. But hey, you gave us permission to edit it! If you need instructions on how to fix it after this PR is merged (or closed) let us know.

BigWhale commented:
> Never had this issue.... what version of ubuntu? Description also doesn't make much sense...
> old code checks all sub directories in /media/ ie goes through each users directories looking for REARM new code assembles the directory "/"+"media"+"/"+{username} and checks for REARM... This is the same thing with less directory trawling...

You're right. Description is wrong. When I noticed that the script isn't working as it should be I just rewrote that part.

There are a couple of symlinks in the /media directory and os.listdir will throw an exception if you feed something that isn't a file to it, so the script will fail in the line 82. Also, listdir throws an exception if it can't open a directory because of the permissions. So it would fail if at any point another user mounted any drives. User directory will stay in /media even if nothing is mounted by that user.

Now, script will check just current user's sub directory so there should be no problems with permissions and all the other exceptions will be handled by the existing try/except. I also added a break if target directory/file was found because there's no point in checking other mounted drives.

BigWhale commented:
> Rebased. Sorry, we had to screw up your `2.0.x` branch to do it. But hey, you gave us permission to edit it! If you need instructions on how to fix it after this PR is merged (or closed) let us know.

That's ok, no need to worry. I made this branch just for the PR. :)
